webapp/channels/src/components/user_settings/display/render_emoticons_as_emoji/index.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import type {Dispatch} from 'redux';
+
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {get} from 'mattermost-redux/selectors/entities/preferences';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+
+import {Preferences} from 'utils/constants';
+
+import type {GlobalState} from 'types/store';
+
+import RenderEmoticonsAsEmoji from './render_emoticons_as_emoji';
+import type {OwnProps} from './render_emoticons_as_emoji';
+
+export function mapStateToProps(state: GlobalState, props: OwnProps) {
+    return {
+        userId: props.adminMode ? props.userId : getCurrentUserId(state),
+        renderEmoticonsAsEmoji: get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.RENDER_EMOTICONS_AS_EMOJI, Preferences.RENDER_EMOTICONS_AS_EMOJI_DEFAULT),
+    };
```

Comment:
Reading over https://github.com/mattermost/mattermost/pull/29414#discussion_r1866070742 again, I realize we both misunderstood what the code was doing here before. The way that the preferences code works when you're an admin editing another user's settings is a bit confusing, so what you had before was correct.

```suggestion
    const userPreferences = props.adminMode && props.userPreferences ? props.userPreferences : undefined;

    return {
        userId: props.adminMode ? props.userId : getCurrentUserId(state),
        renderEmoticonsAsEmoji: get(state, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.RENDER_EMOTICONS_AS_EMOJI, Preferences.RENDER_EMOTICONS_AS_EMOJI_DEFAULT, userPreferences),
    };
```